### PR TITLE
feat: add local postgres container and create databaes client

### PR DESCRIPTION
**Pull request by kiranjthomas**
@geramirez this is a proof of concept for adding Postgres

### Description of changes
* add docker container for postgres
* add docker-compose to easily start Postgres container; I added docker-compose because I think it would be nice to start the whole project with docker-compose in the future
* create knex package to handle migrations and create database client that can be imported by the server
* add db_status endpoint to display db connectivity
* add knex README
* update root README with instructions for starting Postgres

If you like where this is headed, I'll merge and create a new PR to update the StatusRepository with Postgres and remove lokijs.

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "@mueller-splash-pad/knex": "file:../knex",
```
Comment (kiranjthomas) [inline server/package.json:18]: is it ok to import a package this way? I didn't want to publish the private package. I'm not a npm expert :(

Comment (geramirez) [reply server/package.json:18]: yes! totally fine. It's the new way of importing local packages.

**Comment by geramirez** [comment]
Looks like a good setup! I think we can merge this in without affecting anything else. Let me know if I can help out with anything. I think the only change that we'll need is in the pacakge.json in the root folder. 

We'll have to add something like this to the heroku-postbuild. 
```
    "heroku-postbuild": "cd knex && npm i && cd .. && cd client && npm install && npm run build && cd ../server && npm install"

```

**Comment by kiranjthomas** [comment]
Cool! I'll make adjustments for the heroku-postbuild in a subsequent PR.

**Pull request merged**